test/shortestpaths/spfa.jl

```diff
@@ -0,0 +1,176 @@
+            z = @inferred(spfa_shortest_paths(g, 1))
+            @test z == [0, inf, inf]
+        end
+    end
+
+    @testset "Random Graphs" begin
+        @testset "Simple graphs" begin
+            for i = 1:5
+                nvg = Int(ceil(250*rand()))
+                neg = Int(floor((nvg*(nvg-1)/2)*rand()))
+                seed = Int(floor(100*rand()))
+                g = SimpleGraph(nvg, neg; seed = seed)
+                z = spfa_shortest_paths(g, 1)
+                y = dijkstra_shortest_paths(g, 1)
+                @test isapprox(z, y.dists)
+            end
+        end
+
+        @testset "Simple DiGraphs" begin
+            for i = 1:5
+                nvg = Int(ceil(250*rand()))
+                neg = Int(floor((nvg*(nvg-1)/2)*rand()))
+                seed = Int(floor(100*rand()))
+                g = SimpleDiGraph(nvg, neg; seed = seed)
+                z = spfa_shortest_paths(g, 1)
+                y = dijkstra_shortest_paths(g, 1)
+                @test isapprox(z, y.dists)
+            end
+        end
+    end
+
+    @testset "Different types of graph" begin
+        G = CompleteGraph(9)
+        z = spfa_shortest_paths(G, 1)
+        y = dijkstra_shortest_paths(G, 1)
+        @test isapprox(z, y.dists)
+
+        G = CompleteDiGraph(9)
+        z = spfa_shortest_paths(G, 1)
+        y = dijkstra_shortest_paths(G, 1)
+        @test isapprox(z, y.dists)
+
+        G = CycleGraph(9)
+        z = spfa_shortest_paths(G, 1)
+        y = dijkstra_shortest_paths(G, 1)
+        @test isapprox(z, y.dists)
+
+        G = CycleDiGraph(9)
+        z = spfa_shortest_paths(G, 1)
+        y = dijkstra_shortest_paths(G, 1)
+        @test isapprox(z, y.dists)
+
+        G = StarGraph(9)
+        z = spfa_shortest_paths(G, 1)
+        y = dijkstra_shortest_paths(G, 1)
+        @test isapprox(z, y.dists)
+
+        G = WheelGraph(9)
+        z = spfa_shortest_paths(G, 1)
+        y = dijkstra_shortest_paths(G, 1)
+        @test isapprox(z, y.dists)
+
+        G = RoachGraph(9)
+        z = spfa_shortest_paths(G, 1)
+        y = dijkstra_shortest_paths(G, 1)
+        @test isapprox(z, y.dists)
+
+        G = CliqueGraph(5, 19)
+        z = spfa_shortest_paths(G, 1)
+        y = dijkstra_shortest_paths(G, 1)
+        @test isapprox(z, y.dists)
+
+        @testset "Small Graphs" begin
+            for s in [:bull, :chvatal, :cubical, :desargues,
+                      :diamond, :dodecahedral, :frucht, :heawood,
+                      :house, :housex, :icosahedral, :krackhardtkite, :moebiuskantor,
+                      :octahedral, :pappus, :petersen, :sedgewickmaze, :tutte,
+                      :tetrahedral, :truncatedcube, :truncatedtetrahedron, :truncatedtetrahedron_dir]
+                G = smallgraph(s)
+                z = spfa_shortest_paths(G, 1)
+                y = dijkstra_shortest_paths(G, 1)
+                @test isapprox(z, y.dists)
+            end
+        end
+    end
+
+    # Negative Cycle
```

Comment:
You could've made a testset for this, and try using parameterized testsets as per #1184